orttraining/orttraining/python/training/ortmodule/ortmodule.py

```diff
@@ -269,3 +281,39 @@ def named_modules(self, *args, **kwargs):
         """Override :meth:`~torch.nn.Module.named_modules`"""
 
         yield from self._torch_module.named_modules(*args, **kwargs)
+
+    def __getattr__(self, name: str):
+        if '_torch_module' in self.__dict__:
+            # If attribute is not found in ORTModule, it must be present in the
+            # user's torch.nn.Module. Forward the call to the user's model.
```

Comment:
I didnt see `name` being used in this `if` condition, so this comment seems to be outdated. My undertsanding is that this `if` checks whether `self` is ORTModule, because only it should have `_torch_module` (although user model could have it, by coincidence too)

Reply:
```__getattr__``` only gets called when python could not find any attribute in ```self``` that matches with ```name```. So, what that means is that if this method is called, then the attribute does not exist on ```ORTModule```. So, it has to either exist in the user's model or it does not exist at all. So, we just forward the call to the user's model and if it is not defined there, it will throw an ```AttributeError```.